src/dev-app/mdc-slide-toggle/mdc-slide-toggle-demo.html

```diff
@@ -4,7 +4,7 @@
     Default Slide Toggle
   </mat-slide-toggle>
 
-  <mat-slide-toggle [(ngModel)]="firstToggle" disabled>
+  <mat-slide-toggle [(ngModel)]="firstToggle" [disabled]="true">
```

Comment:
Wasn't the whole point of those coercion types to allow usages like this?

Reply:
Yeah, but in some cases where multiple directives are used on the same element, the union input type is used.  In that case, the union of `NgModel#disabled` and `MatSlideToggle#disabled` is `boolean`.

Reply:
I see. Shouldn't we fix `NgModel` in that case?

Reply:
I brought this up in the past when I disabled this flag/when I tried early type checking, but I think we didn't decide on an action.  My thinking is that this is not an issue of `NgModel`. NgModel does not do coercion and ngtsc correctly reports that an empty string is not assignable.

_Though_, we could say that this is such a common case in `NgModel` and we should do it for convenience. Happy to send a PR if so.

Ultimately though this feels like a general problem not specific to only `NgModel`. There might be more other inputs in framework, or in third party libraries that cause these problems. This might be another reason for having something like `@BooleanInput` in the future. cc. @jelbourn

Reply:
I would love to change `NgModel` to follow our model for `disabled`, but it _is_ a breaking change and I think it's too late to make this change in v9. We just have to deal with it for now and file an issue to make the fix for v10.

Reply:
I actually just read about https://github.com/angular/angular/issues/19929 and it looks like `NgModel` already does coercion. So it seems like it would be the right thing to add a static coercion member?

Reply:
So it does: https://github.com/angular/angular/blob/ac34a1429be08e05b110b87f8224d0c25e61a0ef/packages/forms/src/directives/ng_model.ts#L331

We should add the acceptance type, then, and then remove the `disable` bits here

Reply:
Yeah. I created https://github.com/angular/angular/pull/34438

Reply:
https://github.com/angular/angular/pull/34502 😞